File: Haskell/TwoSum.hs
import Data.List (elemIndex)
import Data.Maybe (fromJust)

main:: IO()
main = case result of
    Just (a, b) -> putStrLn $ show a ++ "," ++ show b
    Nothing -> putStrLn "Nothing"
    where
        result :: Maybe (Int, Int)
        result = test [2,3,5,7,11] 9

test :: [Int] -> Int -> Maybe (Int, Int)
test nums target = (\a b -> (div len a, mod len b)) <$> temp <*> temp
-- test nums target = (,) <$> (mod len <$> temp) <*> (div len <$> temp)
-- test nums target = (mod len <$> temp, div len <$> temp)
    where
        step1 :: [Int]
        step1 = (+) <$> nums <*> nums
        temp :: Maybe Int
        temp = elemIndex target step1
        len :: Int
        len = length nums

File: CPP/TwoSum.cpp
#include <vector>
#include <iostream>

using namespace std;
vector<int> twoSum(vector<int>& nums, int target);


int main(){
    vector<int> mynumbs = {2, 7, 11, 15};
    int mytarget = 9;

    vector<int> result = twoSum(mynumbs, mytarget);

    if (result.size() == 2) {
        cout << "Indices found: " << result[0] << ", " << result[1] << endl;
    } else {
        cout << "No solution found." << endl;
    }

    return 0;
}

vector<int> twoSum(vector<int>& nums, int target){
    for(int i = 0; i < nums.size(); i++){
        for(int j = i+1; j < nums.size(); j++){
            if(nums[i]+nums[j] == target){
                return {i, j};
            }
        }
    }
    return {-1};
}